content/lessons/genesis/messages.js

```diff
@@ -0,0 +1,11 @@
+import { defineMessages } from 'react-intl'
+import { formatMessages } from 'lib/messages'
+
+const definedMessages = defineMessages({
+  blockExplorer: 'block explorer',
+  blockExplorerTooltipContent: 'We recommend https://blockstream.info/',
+  firstParagraph:
+    'Let’s find the very first block in the Bitcoin blockchain. Go to a {blockExplorerTooltip} and search for ’0’. This will take you to the block at height 0 you can see, this block only contains one transaction.',
+})
+
+export default formatMessages(definedMessages, 'content.lessons.genesis')
```

Comment:
The 2nd param  needs to match the path to the folder of the messages.js file.

Reply:
I feel there is a little too much boilerplate in this file. It would be great if we could have it be JSON instead and handle the JS elsewhere. This would make it easier for contributors who are not familiar with `react-intl` to add translations and gets rid of the .js/.ts issue. I am not familiar with `react-intl` though, so please let me know if this is possible.

Reply:
Usually you don't need this extra util. It should be possible to do things with existing message extraction tooling, but I couldn't figure out how to get any of them to work with Next.js. Not sure if it's worth the time to continue trying to figure out how to get it to work. I spent hours trying to figure it out yesterday and ended up just writing this hack instead. Another option is to ditch existing message extraction tooling and write our own message extraction script, but I'm not sure if that's worth our time either. Maybe you or someone else want to try and take a quick stab at trying to figure out how we could remove this hack.

If we keep my hack, we could split the message definitions and formatting of the messages up into different files, but I actually think it's better to couple them like this. It's more convenient and the contributors writing this code will most likely be the same people writing the components that consume this data, so they'd have to become familiar with react-intl. Translators shouldn't ever be touching these files. They will only work with the JSON files that get created by the message extraction tooling or they would work with the platform that we upload the JSON files to. 

I'd suggest trying out this pattern and if it doesn't feel right after creating a few messages.js files, then we pivot.

Reply:
> the contributors writing this code will most likely be the same people writing the components that consume this data, so they'd have to become familiar with react-intl

This is kind of what I'm trying to prevent though. I feel contributors shouldn't have to learn how `react-intl` works just to add some text to their challenge. I'm fine with keeping this as-is for now, but I do think we need to simplify this a little further later on. Ideally contributors would just create a `messages.json/translations.json` file, add some strings and would then be able to use them in their components by referencing the key, like in my solution.